numba/tests/test_array_reductions.py

```diff
@@ -665,6 +668,93 @@ def test_nancumsum_basic(self):
         self.check_cumulative(array_nancumsum)
         self.check_nan_cumulative(array_nancumsum)
 
+    def test_ptp_basic(self):
+        pyfunc = array_ptp_global
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a):
+            expected = pyfunc(a)
+            got = cfunc(a)
+            self.assertPreciseEqual(expected, got)
+
+        def a_variations():
+            yield np.arange(10)
+            yield np.array([-1.1, np.nan, 2.2])
+            yield np.array([-np.inf, 5])
+            yield (4, 2, 5)
+            yield (1,)
+            yield np.full(5, 5)
+            yield [2.2, -2.3, 0.1]
+            a = np.linspace(-10, 10, 16).reshape(4, 2, 2)
+            yield a
+            yield np.asfortranarray(a)
+            yield 6
+            yield 6.5
+            yield -np.inf
+            yield 1 + 4j
+
+        for a in a_variations():
+            check(a)
+
+    def test_ptp_complex(self):
+        pyfunc = array_ptp_global
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a):
+            expected = pyfunc(a)
+            got = cfunc(a)
+            self.assertPreciseEqual(expected, got)
+
+        def make_array(real_nan=False, imag_nan=False):
+            real = np.linspace(-4, 4, 25)
+            if real_nan:
+                real[4:9] = np.nan
+            imag = np.linspace(-5, 5, 25)
+            if imag_nan:
+                imag[7:12] = np.nan
+            return (real + 1j * imag).reshape(5, 5)
+
+        comp = make_array()
+        check(comp)
+
+        comp = make_array(real_nan=True)
+        check(comp)
+
+        comp = make_array(imag_nan=True)
+        check(comp)
+
+        comp = make_array(real_nan=True, imag_nan=True)
+        check(comp)
+
+        real = np.ones(8)
+        imag = np.arange(-4, 4)
+        comp = real + 1j * imag
+        check(comp)
+        comp = real - 1j * imag
+        check(comp)
+
+        comp = np.full((4, 4), fill_value=(1 - 1j))
+        check(comp)
```

Comment:
Is there any testing of scalar complex values?

Reply:
I'm not sure what you mean.

Reply:
I think I've just spotted some I missed above, was looking for e.g. `x + y * j` as input.